### DISCUSSION TITLE
Perception tracking isssue

### DISCUSSION THREAD

**Discussion by Sevenfortyseven** [pointfreeco/swift-perception#60] [General]
### Description

Even tho I'm not using @Perception framework and also my view is embedded inside WithPerceptionTracking block, I'm still getting 
`Perceptible state was accessed but is not being tracked. Track changes to state by wrapping your view in a 'WithPerceptionTracking' view.` warning. Warning appears when this view is presented, and it also generates warnings when I enter characters in a textfield and simulator becomes unresponsive. what can be an issue here?
![Screenshot 2024-03-03 at 14 43 43](https://github.com/pointfreeco/swift-perception/assets/86253061/069ceb38-18ae-4469-a5d4-1f42e30b8e30)
![Screenshot 2024-03-03 at 14 44 37](https://github.com/pointfreeco/swift-perception/assets/86253061/bd0224ef-e85f-4f9e-8a47-e550b3852d8b)

package target is the main branch of the project

### Checklist

- [X] I have determined that this bug is not reproducible using Swift's observation tools. If the bug is reproducible using the `@Observable` macro or another tool from the `Observation` framework, please [file it directly with Apple](https://github.com/apple/swift/issues/new/choose).
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

this code to work smoothly

### Actual behavior

I'm getting unexpected warnings and simulator becomes unresponsive while I'm typing in textfield

### Destination operating system

iOS 14

### Xcode version information

5.1

**Comment by stephencelis** [comment]
@Sevenfortyseven I'm not sure I understand. Can you zip up an Xcode project that demonstrates the issue and attach it here?

**Comment by mbrandonw** [comment]
Hi @Sevenfortyseven, I'm going to convert this to a discussion for now. Please feel free to share your project in the discussion if you are able to reproduce for us.

**Comment by Sevenfortyseven** [comment]
@mbrandonw  HI Brandon. I'm not able to reproduce this issue any more. Thank you!
@stephencelis Sorry, I somehow missed your comment, but hopefully everything works perfectly by now

**Comment by jerometonnelier** [comment]
Hi guys :-) (@mbrandonw @stephencelis @Sevenfortyseven )
Just for the record, since last Xcode/TCA update I get theses errors a lot. Here is a project that I used that gets the error (I must be doing something wrong I'm pretty sure), but since last update, the menu is reaaaaaaaaally slow compared to what it used to. 
Thanks for your feedback :-) 
[MobileMenutest.zip](https://github.com/user-attachments/files/17688421/MobileMenutest.zip)

**Reply by mbrandonw** [reply]
Hi @jerometonnelier, this project is targeting iOS 16 and using `@ObservableState` but is not using `WithPerceptionTracking` anywhere. So this project is not using the tools correctly. Have you read the documentation to see how these tools are meant to be used? Also the warning printed to the console mentions what the fix is too ("Track changes to state by wrapping your view in a 'WithPerceptionTracking' view.").

**Reply by jerometonnelier** [reply]
Hi @mbrandonw  and thanks for the answer. Actually I tried to update an "old" TCA project to a more modern syntax. I saw the @Reducer and @ObservableState stuff, but nothing that stipulates the use of WithPerceptionTracking that should go along with it. I'll try to have a look for it in the documentation. Sorry for the disturbance 😓

**Reply by mbrandonw** [reply]
If you followed the migration guides you will find the information in [this one](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/migratingto1.7). And there's a dedicated article [here](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/observationbackport).

**Reply by jerometonnelier** [reply]
wow thanks a lot, that's so kind of you ❤️ I'll have a look ^^
